src/components/CytoscapeGraph/Layout/GroupCompoundLayout.ts

```diff
@@ -107,117 +110,119 @@ export default class GroupCompoundLayout {
    * This code gets executed on the cy.layout(...).run() is our entrypoint of this algorithm.
    */
   run() {
-    const { realLayout, compoundLayoutOptions } = this.options;
-    const parents = this.parents();
-    const children = parents.children();
-
-    // (1.a) Prepare parents by assigning a size and running the compound layout
-    parents.each(parent => {
-      const children = parent.children();
-      const targetElements = children.add(children.edgesTo(children));
-
-      // We expect a discrete layout here
-      const compoundLayout = targetElements.layout(compoundLayoutOptions);
-      compoundLayout.on('layoutstart layoutready layoutstop', _evt => {
-        // Avoid to propagate any local layout events up to cy, this would yield a global operation when not all nodes are ready.
-        return false;
-      });
-      compoundLayout.run();
+    [BoxByType.APP, BoxByType.NAMESPACE, BoxByType.CLUSTER].forEach(boxByType => {
+      const { realLayout, compoundLayoutOptions } = this.options;
+      const parents = this.parents(boxByType);
+      const children = parents.children();
 
-      // see https://github.com/cytoscape/cytoscape.js/issues/2402
-      const boundingBox = parent.boundingBox();
+      // (1.a) Prepare parents by assigning a size and running the compound layout
+      parents.each(parent => {
+        const children = parent.children();
+        const targetElements = children.add(children.edgesTo(children));
+
+        // We expect a discrete layout here
+        const compoundLayout = targetElements.layout(compoundLayoutOptions);
```

Comment:
This is `dagre` by default IIRC. So we are running that as the layout for any child.

Reply:
Yes, it is `dagre` https://github.com/kiali/kiali-ui/blob/master/src/components/CytoscapeGraph/CytoscapeGraphUtils.ts#L111

Reply:
Right, I think using Dagre inside the boxes is a good idea.  The "real" layout would apply to the final layout of the compound nodes, and nodes with no parent, right?

Reply:
Right, to every "top" node.